src/app/file/pixly_format.cpp

```diff
@@ -0,0 +1,535 @@
+    }
+
+    std::vector<int> visible(layerCount, 0);
+
+    TiXmlElement* xmlFrame = check(xmlFrames->FirstChild("Frame"))->ToElement();
+    while (xmlFrame) {
+      TiXmlElement* xmlRegion = check(xmlFrame->FirstChild("Region"))->ToElement();
+      TiXmlElement* xmlIndex = check(xmlFrame->FirstChild("Index"))->ToElement();
+
+      int index = check_number<int>(xmlIndex->Attribute("linear"));
+      frame_t frame(index / layerCount);
+      LayerIndex layer_index(index % layerCount);
+      Layer *layer = sprite->indexToLayer(layer_index);
+
+      const char * duration = xmlFrame->Attribute("duration");
+      if(duration) {
+        sprite->setFrameDuration(frame, base::convert_to<int>(std::string(duration)));
+      }
+
+      visible[(int)layer_index] += (int)(std::string(check(xmlFrame->Attribute("visible"),"false")) == "true");
+
+      int x0 = check_number<int>(xmlRegion->Attribute("x"));
+      int y0 = check_number<int>(xmlRegion->Attribute("y")); // inverted
+
+      if(y0 < 0 || y0 + frameHeight > imageHeight || x0 < 0 || x0 + frameWidth > imageWidth) {
+        throw Exception("looking for cels outside the bounds of the PNG");
+      }
+
+      base::UniquePtr<Cel> cel;
+      ImageRef image(Image::create(pixelFormat, frameWidth, frameHeight));
+
+      // Convert rows_pointer into the doc::Image
+      for (int y = 0; y < frameHeight; y++) {
+        // RGB_ALPHA
+        uint8_t* src_address = rows_pointer[imageHeight-1 - y0 - (frameHeight-1) + y] + (x0 * 4);
+        uint32_t* dst_address = (uint32_t*)image->getPixelAddress(0, y);
+        unsigned int r, g, b, a;
+
+        for (int x=0; x<frameWidth; x++) {
+          r = *(src_address++);
+          g = *(src_address++);
+          b = *(src_address++);
+          a = *(src_address++);
+          *(dst_address++) = rgba(r, g, b, a);
+        }
+      }
+
+      cel.reset(new Cel(frame, image));
+      static_cast<LayerImage*>(layer)->addCel(cel);
+      cel.release();
+
+      xmlFrame = xmlFrame->NextSiblingElement();
+      fop->setProgress(0.75 + 0.25 * ((float)index / (float)imageCount));
+    }
+
+    for(int i=0; i<layerCount; i++) {
+      LayerIndex layer_index(i);
+      Layer *layer = sprite->indexToLayer(layer_index);
+      layer->setVisible(visible[i] > frameCount/2);
+    }
+
+    fop->createDocument(sprite);
+    sprite.release();
+  }
+  catch(Exception &e) {
+    fop->setError((std::string("Pixly file format: ")+std::string(e.what())+"\n").c_str());
+    success = false;
+  }
+
+  for (y = 0; y < height; y++) {
+    png_free(png_ptr, rows_pointer[y]);
+  }
+  png_free(png_ptr, rows_pointer);
+
+  // Clean up after the read, and free any memory allocated
+  png_destroy_read_struct(&png_ptr, &info_ptr, NULL);
+  return success;
+}
+
+#ifdef ENABLE_SAVE
+bool PixlyFormat::onSave(FileOp* fop)
+{
+  const Sprite* sprite = fop->document()->sprite();
+
+  if(sprite->pixelFormat() != IMAGE_RGB || !fop->document()->sprite()->needAlpha()) {
```

Comment:
There is no need to check `needAlpha`. Also, you can remove the `pixelFormat()` check if you remove the `FILE_SUPPORT_GRAY `, `FILE_SUPPORT_GRAYA`, and `FILE_SUPPORT_INDEXED` from `onGetFlags()` (and you should remove those constants from `onGetFlags()` as this format doesn't support Grayscale, Indexed)